app/src/main/java/com/android/feedme/model/data/RecipeRepository.kt

```diff
@@ -107,32 +134,53 @@ class RecipeRepository(private val db: FirebaseFirestore) {
       onSuccess: (List<Recipe>, DocumentSnapshot?) -> Unit,
       onFailure: (Exception) -> Unit
   ) {
+    // Prefix search query
     var queryRef =
         db.collection(collectionPath)
-            .whereGreaterThanOrEqualTo("title", query)
-            .whereLessThan("title", query + "\uf8ff")
-            .limit(10) // Limit the number of documents fetched
+            .whereArrayContainsAny("searchItems", listOf(query.lowercase(Locale.ROOT)))
+            .limit(6) // Limit the number of documents fetched
```

Comment:
Why did we choose a limit of 6 ?

Reply:
Also it would be nice to explain what the query does exactly with a comment

Reply:
Recipe cards take enough space in the UI so a limit of 6 is more reasonable than 10 (before the user can press on load more)